src/main/scala/com/ambiata/poacher/hdfs/HdfsTemporary.scala

```diff
@@ -0,0 +1,47 @@
+package com.ambiata.poacher.hdfs
+
+import com.ambiata.mundane.control._
+import com.ambiata.mundane.io._
+import com.ambiata.mundane.io.Temporary._
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import java.util.concurrent.atomic.AtomicInteger
+import scalaz._, Scalaz._
+
+case class HdfsTemporary(seed: String) {
+  private val step: AtomicInteger = new AtomicInteger(0)
+
+  def path: Hdfs[Path] = {
+    val base = uniqueDirPath
+    val incr = step.incrementAndGet.toString
+    val path = new Path(new Path(base.path, incr), seed)
+    val msg = s"Hdfs($path)"
+    addCleanupFinalizer(new Path(base.path), msg) >>
+      addPrintFinalizer(msg) >>
+        Hdfs.ok(path)
+  }
+
+  def addCleanupFinalizer(path: Path, msg: String): Hdfs[Unit] =
+    if (skipCleanup) forceAddPrintFinalizer(msg)
```

Comment:
Adding this here seems a little weird. Should the message be different? /notsure